test/units/testsuite-21.sh

```diff
@@ -8,22 +8,16 @@ set -o pipefail
 systemctl list-jobs | grep -F 'end.service' && SHUTDOWN_AT_EXIT=1 || SHUTDOWN_AT_EXIT=0
 
 at_exit() {
-    # "Safety net" - check for any coredumps which might have not caused dfuzzer
-    # to stop & return an error (we need to do this now before truncating the
-    # journal)
-    # TODO: check fo ASan/UBSan errors
-    local found_cd=0
-    while read -r exe; do
-        coredumctl info "$exe"
-        found_cd=1
-    done < <(coredumpctl -F COREDUMP_EXE | sort -u)
-    [[ $found_cd -eq 0 ]] || exit 1
-
-    # We have to call the end.service explicitly even if it's specified on
+    set +e
+    # We have to call the end.service/poweroff explicitly even if it's specified on
     # the kernel cmdline via systemd.wants=end.service, since dfuzzer calls
     # org.freedesktop.systemd1.Manager.ClearJobs() which drops the service
     # from the queue
-    [[ $SHUTDOWN_AT_EXIT -ne 0 ]] && systemctl start --job-mode=flush end.service
+    if [[ $SHUTDOWN_AT_EXIT -ne 0 ]]; then
+        systemctl poweroff
+        sleep 15
+        systemctl poweroff -ff
```

Comment:
I think it can be replaced with something like
```
if ! systemctl poweroff; then
    # PID1 is down let's try to save the journal
    journalctl --sync || true # journal can be down as well so let's ignore exit codes here
    systemctl -ff poweroff # sync() and reboot(RB_POWER_OFF)
fi
```
It should probably cover the worst scenario where `dbus` is down, `systemd` is frozen and `journal` isn't responsive (hopefully it shouldn't happen very often these days)